src/node/reconfig_id.h

```diff
@@ -8,20 +8,42 @@
 
 namespace ccf
 {
-  inline size_t get_next_reconfiguration_id(
-    ccf::NetworkState& network, kv::ReadOnlyTx& tx)
+  inline kv::ReconfigurationId get_next_reconfiguration_id(
+    ccf::NetworkTables& tables, kv::ReadOnlyTx& tx)
   {
-    auto nodes = tx.ro(network.nodes);
+    auto nconfigs = tx.ro(tables.network_configurations);
+    auto e = nconfigs->get((kv::ReconfigurationId)-1);
+    if (!e.has_value())
+    {
+      return 0;
+    }
+    else
+    {
+      return e.value().rid + 1;
+    }
+  }
 
-    size_t max_id = 0;
-    nodes->foreach([&max_id](const auto& node_id, const auto& node_info) {
-      if (node_info.reconfiguration_id.has_value())
-      {
-        max_id = std::max(max_id, node_info.reconfiguration_id.value());
-      }
-      return true;
-    });
+  inline kv::NetworkConfiguration get_latest_network_configuration(
+    ccf::NetworkTables& tables, kv::Tx& tx)
+  {
+    auto nconfigs = tx.ro(tables.network_configurations);
+    auto e = nconfigs->get((kv::ReconfigurationId)-1);
+    if (e.has_value())
+    {
+      return nconfigs->get(e.value().rid).value();
+    }
+    return {};
+  }
 
-    return max_id + 1;
+  inline void add_new_network_reconfiguration(
+    ccf::NetworkTables& tables, kv::Tx& tx, kv::NetworkConfiguration& config)
+  {
+    config.rid = get_next_reconfiguration_id(tables, tx);
+    LOG_DEBUG_FMT(
+      "Configurations: adding new entry to network_configurations table: {}",
+      config);
+    auto nconfigs = tx.rw(tables.network_configurations);
+    nconfigs->put(config.rid, config);
+    nconfigs->put((kv::ReconfigurationId)-1, {config.rid, {}});
```

Comment:
I'm a little confused about this. AIUI, we keep track of the latest config id at a well-known key (`-1`) and then we store the config at its id. If that's what we want to do, we should at least add a comment next to the table definition to explain this. This will also cause memory growth in the KV store as more nodes are added to the service. Is this intended?

Reply:
Sure, will add a comment for `-1`.  And yes, outdated configurations will be removed when we switch to them, so that only the current one and future entries are retained. This isn't in this PR yet, because I couldn't find a way to correlate between old config (nodes table, no ID) and new config (with ID in network_configs) yet, so at the point where we switch to the next config, I don't know the right ID.